core-processor/src/executor.rs

```diff
@@ -488,3 +499,148 @@ pub fn execute_wasm<
         allocations: info.allocations,
     })
 }
+
+#[cfg(test)]
+mod tests {
```

Comment:
I'd also add test for pages to be updated with and without lazy pages feature

Reply:
Added them.